static/app/views/ddm/chart/useMetricChartSamples.tsx

```diff
@@ -133,7 +133,7 @@ export function useMetricChartSamples({
 
   const handleClick = useCallback<EChartClickHandler>(
     (event: EChartMouseEventParam) => {
-      const sample = samplesById[event.seriesId];
+      const sample = samplesById[event.seriesName];
```

Comment:
The id now contains the sample index to dedupe ids so we were always getting `undefined` here